packages/@react-aria/overlays/src/calculatePosition.ts

```diff
@@ -340,16 +340,24 @@ export function calculatePositionInternal(
 
   let arrowPosition: Position = {};
 
-  const preferredArrowPosition = overlaySize[crossSize] / 2;
-
-  const arrowOverlappingChildMinEdge = childOffset[crossAxis] - overlaySize[crossAxis] + (arrowCrossSize / 2);
-  const arrowOverlappingChildMaxEdge = childOffset[crossAxis] + childOffset[crossSize] - overlaySize[crossAxis] - (arrowCrossSize / 2);
+  // All coords are transformed so that (0, 0) is at the top left of the overlay
+  // Prefer the arrow being in the center of the trigger/overlay anchor element
+  let preferredArrowPosition = childOffset[crossAxis] + .5 * childOffset[crossSize] - overlaySize[crossAxis];
```

Comment:
why `- overlaySize[crossAxis]`? seems like the preferred ArrowPosition has nothing to do with the overlay now

Reply:
Still need arrowOverlappingChild... variables i think
edit: looks like you moved them down below, but lets keep them here

Reply:
So the `- overlaySize[crossAxis]` is used to transform the coordinates in such a way that they are w/ respect to the overlay rather than being with respect to the document (i.e. 0 now refers to the top/left of the overlay instead of the top/left of the document/boundary element). This is done since the arrow is `position: absolute` inside the overlay div itself, hence we need a value that uses the top/left of the overlay as the origin.